### PR TITLE
refactor: expanded cli version info

### PR DESCRIPTION
**Pull request by onbjerg**
There were some issues with https://github.com/paradigmxyz/reth/pull/2277, mostly that it creates a function (that is not used anywhere), which we cannot use for Clap as it expects a string literal. We could use the function if we used the builder API, but we don't.

This PR:

- Creates 2 constants for version info (short and long)
- Applies these constants to Clap and the initial log in the commands

The short version:

```
reth 0.1.0 (defa64b2)
```

The long version:

```
reth Version: 0.1.0
Commit SHA: defa64b2
Build Timestamp: 2023-05-19T01:47:19.815651705Z
Build Features:
```

Originally I intended to display the build profile as opposed to the build features, but Cargo `PROFILE` sets any profile that isn't explicity `release` to `debug` (and we want to release with the `maxperf` profile)

### CODE REVIEW

File: bin/reth/src/version.rs [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
+/// The long version information for reth.
+///
+/// Expands to the short version information for reth.
+///
+/// - The latest version from Cargo.tom
+/// - The long SHA of the latest commit.
+/// - The build datetime
+/// - The build features
+///
+/// Example: "1.2.3 (defa64b2)"
```
Comment (mattsse) [inline bin/reth/src/version.rs:21]: this example is the same as above.
should be `LONG_VERSION`

File: bin/reth/src/version.rs [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
+pub(crate) const SHORT_VERSION: &str =
+    concat!(env!("CARGO_PKG_VERSION"), " (", env!("VERGEN_GIT_SHA"), ")");
 
-/// The version message for the current program, like
-/// `reth/v0.1.0/macos-6fc95a5/`
-pub fn version_message() -> String {
-    format!("{}/v{}/{}-{}", NAME, VERSION, OS, SHA.unwrap())
-}
```
Comment (mattsse) [inline bin/reth/src/version.rs:13]: I know this isn't used anywhere, because I didn't integrate this yet, but this was intended for the client version that's being used in the p2p messages, which should also be customizable via cli arg.

this is not a dealbreaker for this PR, I'll just open an issue for this separately.

**Review by mattsse — APPROVED** [review]
lgtm, smol doc example nit
